### PR TITLE
Basic Search in projects page

### PR DESCRIPTION
**Pull request by bansalnagesh**
1. Need help with the styling of the search bar. 
2. The search as of now is a single record search and hence does not include skill based search. Will include that in a follow-up diff.

### CODE REVIEW

**Comment by bansalnagesh** [comment]
Related to the first part of Issue #18

**Comment by adrianthedev** [comment]
image

File: app/frontend/svgs/search-outline.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M21 21L15 15M17 10C17 13.866 13.866 17 10 17C6.13401 17 3 13.866 3 10C3 6.13401 6.13401 3 10 3C13.866 3 17 6.13401 17 10Z" stroke="#4A5568" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
```
Comment (sradu) [inline app/frontend/svgs/search-outline.svg]: indentation?

File: Gemfile [hunk 2]
```diff
@@ -76,5 +76,7 @@ gem 'aws-sdk-rails'
 # Exceptions.
 gem 'exception_notification', git: 'https://github.com/smartinez87/exception_notification'
 
+#Search
+gem 'pg_search'
```
Comment (sradu) [inline Gemfile]: a newline after this

File: app/models/project.rb [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 
+  include PgSearch
+
   has_many :volunteers, dependent: :destroy
   has_many :volunteered_users, through: :volunteers, source: :user, dependent: :destroy
 
   acts_as_taggable_on :skills
   acts_as_taggable_on :project_types
 
+  pg_search_scope :search, against: %i(name description participants looking_for location highlight), using: { tsearch: { any_word: true } }
```
Comment (sradu) [inline app/models/project.rb]: Why search in highlight?

File: app/views/projects/index.html.erb [hunk 4]
```diff
@@ -30,10 +30,20 @@
...
+      <nav class="flex justify-between">
         <a href="#" :class="{'bg-gray-200': open}" class="px-3 py-2 font-medium text-sm leading-5 rounded-md text-gray-600 hover:text-gray-800 focus:outline-none focus:text-gray-800 focus:bg-gray-200 fill-current flex-no-wrap" @click.prevent="open = !open" >
           <%= inline_svg_pack_tag('media/svgs/filter-outline.svg', class: 'h-4 inline-block') %> Filter projects
         </a>
+        <% if @show_search_bar %>
+          <div class="w-64">
+            <%= form_with url: projects_path, local: true, method: :get do |form| %>
+              <div class="relative rounded-md shadow-sm">
+                <%= form.text_field :query, placeholder: "Search", value: params[:query], class: "form-input block w-full sm:text-sm sm:leading-5" %>
+                <%= inline_svg_pack_tag('media/svgs/search-outline.svg', class: 'absolute inset-auto right-0 top-0 translate-y-1/2 mr-2 h-4 inline-block', style: 'margin-top: 10px;') %>
```
Comment (sradu) [inline app/views/projects/index.html.erb]: indentation is borked.

Comment (adrianthedev) [reply app/views/projects/index.html.erb]: This is ok. They're siblings.

File: app/models/project.rb (hunk 3 above)
Comment (adrianthedev) [reply app/models/project.rb]: @bansalnagesh?

File: Gemfile (hunk 2 above)
Comment (adrianthedev) [reply Gemfile]: Added `.editorconfig` to take care of this.

**Pull request merged**